src/ximea_ros_cam.cpp

```diff
@@ -206,16 +206,34 @@ void XimeaROSCam::initStorage() {
         boost::filesystem::path calib_dir = main_dir /
                                             (this->cam_name_ + "/calib/");
         this->png_path_ = calib_dir.string();
-        boost::filesystem::create_directories(calib_dir);
+        if (!boost::filesystem::create_directories(calib_dir))
+        {
+            // failed to create directory, exit ROS and explain.
+            ROS_INFO_STREAM("ERROR: unable to create directory: " << this->png_path_);
+            ROS_INFO_STREAM("Please make sure that the image_directory "
+                         << "parameter is set to a folder with the proper "
+                         << "permissions in the config file.");
+            ros::shutdown();
+        }
+
+        this->save_trigger_ = false;
     }
 
     // directory that holds video stream images
-    boost::filesystem::path img_stream_dir = main_dir /
-                                             (this->cam_name_ + "/stream/");
-    this->bin_path_ = img_stream_dir.string();
-    boost::filesystem::create_directories(img_stream_dir);
-
-    this->save_trigger_ = false;
+    if (this->save_disk_) {
+        boost::filesystem::path img_stream_dir = main_dir /
+                                                 (this->cam_name_ + "/stream/");
+        this->bin_path_ = img_stream_dir.string();
```

Comment:
If this is used someplace else is it fine if it is not set?

Reply:
during the streaming process, if it isn't publishing (that is, it is saving in binary image format), it will save into the bin path. So if the save_disk flag is false, bin path should never be used

Reply:
Nice